model/src/main/java/org/projectnessie/api/TreeApi.java

```diff
@@ -144,9 +218,46 @@ LogResponse getCommitLog(
       @NotNull
       @Pattern(regexp = Validation.REF_NAME_OR_HASH_REGEX, message = Validation.REF_NAME_OR_HASH_MESSAGE)
       @Parameter(description = "ref to show log from")
-      @PathParam("ref") String ref)
+      @PathParam("ref")
+          String ref,
+      @Parameter(description = "maximum number of commit-log entries to return, just a hint for the server")
+      @QueryParam("max")
+          Integer maxRecords,
+      @Parameter(description = "pagination continuation token, as returned in the previous LogResponse.token")
+      @QueryParam("pageToken")
+          String pageToken)
           throws NessieNotFoundException;
 
+  /**
+   * Default implementation to return a stream of objects for a ref, functionally equivalent to
+   * calling {@link #getEntries(String, Integer, String)} with manual paging.
+   * <p>The {@link Stream} returned by {@code getEntriesStream(ref, OptionalInt.empty())},
+   * if not limited, returns all commit-log entries.</p>
+   *
+   * @param ref a named reference (branch or tag name) or a commit-hash
+   * @param pageSizeHint page-size hint for the backend
+   * @return stream of {@link Entry} objects
+   */
+  default Stream<Entry> getEntriesStream(@NotNull String ref, OptionalInt pageSizeHint) throws NessieNotFoundException {
```

Comment:
I am not sure why these are in the model. To me they are client only and should be there

Reply:
It's a double-edged sword IMO. The model defines the paging properties via `PaginatedResponse` but doesn't define whether paging is only "legit" via HTTP or whether another tier can actually enforce paging. `NessieClient` "only" exposes the `TreeApi` interface, so we'd either have to add a new `TreeWithConvenienceApi` in `nessie-client` or some class with static utility methods. Both approaches feel a bit odd to me.

Reply:
I actually, personally, don't mind the convenience api. Primarily because the contents tree has a few 'wasteful' apis: delete and put single contents. I thikn there is an argument that the direct jax-rs api shouldn't be exposed, rather act as a delegate so the client can continue to grow convenience methods.